### PR TITLE
Support sanitizing CFF2 table

### LINKED ISSUE #175: Add CFF2 support
Currently ots-sanitize can't recognize CFF2 fonts.
```
% ots-sanitize AdobeVFPrototype.otf
ERROR: no supported glyph shapes table(s) present
Failed to sanitize file!
```

### PR DESCRIPTION
**Pull request by khaledhosny**
Fixes #175.

### CODE REVIEW

File: src/cff.cc [hunk 1]
```diff
@@ -1002,6 +1260,88 @@ OpenTypeCFF::~OpenTypeCFF() {
...
+    return Error("Failed to parse Global Subrs INDEX");
+  }
+
+  // Check if all fd_index in FDSelect are valid.
+  for (const auto& fd_select : this->fd_select) {
+    if (fd_select.second >= this->font_dict_length) {
+      return Error("Invalid FD index: %d >= %d\n",
+                   fd_select.second, this->font_dict_length);
+    }
+  }
```
Comment (cjchapman) [inline src/cff.cc]: Please also add a check that the glyph ranges in the FDSelect do not exceed the number of glyphs in the font.

Comment (khaledhosny) [reply src/cff.cc]: Good idea, done.

File: src/cff_charstring.cc [hunk 2]
```diff
@@ -39,7 +38,16 @@ bool ExecuteCharString(ots::OpenTypeCFF& cff,
                        std::stack<int32_t> *argument_stack,
                        bool *out_found_endchar,
                        bool *out_found_width,
-                       size_t *in_out_num_stems);
+                       size_t *in_out_num_stems,
+                       bool cff2);
```
[the remark this answers is not in the document]
Comment (khaledhosny) [reply src/cff_charstring.cc]: The two callers for this function do the check for CFF2 already. Anyway, the first iteration of this code did not have access to the `OpenTypeCFF` object at all so the `cff2` was passed around, I changed this later but did not see practical benefit for not also passing the boolean.

File: src/cff.h [hunk 3]
```diff
@@ -53,6 +61,27 @@ class OpenTypeCFF : public Table {
...
 
+  // CFF2 VariationStore regionIndexCount.
+  std::vector<uint16_t> region_index_count;
+
+ private:
+  const uint8_t *m_data;
+  size_t m_length;
+};
+
+class OpenTypeCFF2 : public OpenTypeCFF {
```
[the remark this answers is not in the document]
Comment (khaledhosny) [reply src/cff.h]: Ideally, I’d like to refactor this code and make all the separate functions methods of the `OpenTypeCFF` class and do way with passing lots of parameters around and be able to do better error reporting. But I’d like to do this later to minimize the number of changes needed for CFF2 support.

[the remark this answers is not in the document]
Comment (blueshade7) [reply src/cff.h]: That's understandable and fine as long as you are aware of pros & cons.

**Pull request merged**